modules/ags/src/main/scala/lucuma/ags/Ags.scala

```diff
@@ -170,21 +180,23 @@ object Ags {
     wavelength:         Wavelength,
     baseCoordinates:    Coordinates,
     scienceCoordinates: List[Coordinates],
-    position:           AgsPosition,
+    positions:          NonEmptyList[AgsPosition],
     params:             AgsParams
   ): Pipe[F, GuideStarCandidate, AgsAnalysis] = {
 
     // Cache the limits for different speeds
     val guideSpeeds = guideSpeedLimits(constraints, wavelength)
     // This is essentially a cache of geometries avoiding calculatting them
     // over and over again as they don't change for different positions
-    val calcs       = params.posCalculations(List(position))
+    val calcs       = params.posCalculations(positions)
     // use constraints to calculate all guide speeds
     val bc          = constraintsFor(guideSpeeds)
 
     in =>
-      in.filter(c => c.gBrightness.exists(g => bc.exists(_.contains(Band.Gaia, g))))
-        .map { gsc =>
+      (in.filter(c => c.gBrightness.exists(g => bc.exists(_.contains(Band.Gaia, g)))),
+       Stream.emits[F, AgsPosition](positions.toList)
+      )
+        .mapN { (gsc, position) =>
```

Comment:
Ah. Here is the cartesian product you were asking about in slack. :)

Reply:
Yes indeed